<launch>

   <param name='agent_names' value='clara dario'/>
   <param name='agent_colors' value='blue green'/>
   <!-- <arg name='agent_names' value='agent1 agent2 agent3'/> -->
   <!-- <param name='agent_names' value='$(arg agent_names)'/> -->

   <!-- <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/> -->


   <group ns = 'clara'>

    <!-- <param name='deltaT' type='double' value='2.0'/>
    <param name='agentID' type='str' value='agent1'/>
    <param name='delay' type='double' value='0.0'/> -->

    <rosparam param = 'initial_position'>[1.0, 0.0]</rosparam>

    <!-- <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
    <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
    <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/> -->

    <node name='vehicle_simulator'
      pkg='circumnavigation_cloud'
      type='vehicle_simulator.py' output="screen"/>

    <node name='controller'
      pkg='circumnavigation_cloud'
      type='controller.py' output="screen"/>

    <node name='sensor_simulator'
      pkg='circumnavigation_cloud'
      type='sensor_simulator.py' output="screen"/>

   </group>


   <group ns = 'dario'>

    <!-- <param name='deltaT' type='double' value='2.0'/>
    <param name='agentID' type='str' value='agent1'/>
    <param name='delay' type='double' value='0.0'/> -->

    <rosparam param = 'initial_position'>[2.0, 3.0]</rosparam>

    <!-- <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
    <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
    <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/> -->

    <node name='vehicle_simulator'
      pkg='circumnavigation_cloud'
      type='vehicle_simulator.py' output="screen"/>

    <node name='controller'
      pkg='circumnavigation_cloud'
      type='controller.py' output="screen"/>

    <node name='sensor_simulator'
      pkg='circumnavigation_cloud'
      type='sensor_simulator.py' output="screen"/>

   </group>


   <!-- <group ns = 'agent2'>

       <param name='deltaT' type='double' value='2.0'/>

       <param name='agentID' type='str' value='agent2'/>

       <param name='delay' type='double' value='1.0'/>

       <rosparam param = 'initial_position'>[0.0, 1.0]</rosparam>

       <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/>

       <node name='simulator'
        pkg='circumnavigation_cloud'
        type='simulator.py' output="screen"/>

       <node name='controller'
        pkg='circumnavigation_cloud'
        type='controller.py' output="screen"/>

       <node name='sensor_simulator'
        pkg='circumnavigation_cloud'
        type='sensor_simulator.py' output="screen"/>

   </group> -->


   <!-- <group ns = 'agent3'>

       <param name='deltaT' type='double' value='2.0'/>

       <param name='agentID' type='str' value='agent3'/>

       <param name='delay' type='double' value='2.0'/>

       <rosparam param = 'initial_position'>[0.5, 0.5]</rosparam>

       <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/>



       <node name='simulator'
        pkg='circumnavigation_cloud'
        type='simulator.py' output="screen"/>

       <node name='controller'
        pkg='circumnavigation_cloud'
        type='controller.py' output="screen"/>

       <node name='sensor_simulator'
        pkg='circumnavigation_cloud'
        type='sensor_simulator.py' output="screen"/>

   </group> -->

<!--    <group ns = 'agent4'>

       <param name='deltaT' type='double' value='2.0'/>

       <param name='agentID' type='int' value='3'/>

       <param name='delay' type='double' value='2.0'/>

       <rosparam param = 'initial_position'>[1.0, 0.0]</rosparam>

       <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/>



       <node name='simulator'
        pkg='circumnavigation_cloud'
        type='simulator.py' output="screen"/>

       <node name='controller'
        pkg='circumnavigation_cloud'
        type='controller.py' output="screen"/>

       <node name='sensor_simulator'
        pkg='circumnavigation_cloud'
        type='sensor_simulator.py' output="screen"/>

   </group>



   <group ns = 'agent5'>

       <param name='deltaT' type='double' value='2.0'/>

       <param name='agentID' type='int' value='4'/>

       <param name='delay' type='double' value='2.0'/>

       <rosparam param = 'initial_position'>[-1.0, 1.0]</rosparam>

       <rosparam command='load' file='$(find circumnavigation_cloud)/param/target.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/controller.yaml'/>
       <rosparam command='load' file='$(find circumnavigation_cloud)/param/simulator.yaml'/>



       <node name='simulator'
        pkg='circumnavigation_cloud'
        type='simulator.py' output="screen"/>

       <node name='controller'
        pkg='circumnavigation_cloud'
        type='controller.py' output="screen"/>

       <node name='sensor_simulator'
        pkg='circumnavigation_cloud'
        type='sensor_simulator.py' output="screen"/>

   </group> -->

   <!-- <node name='cloud_agent_info'
     pkg='circumnavigation_cloud'
     type='cloud_agent_info.py' output="screen"/>


   <rosparam command='load' file='$(find circumnavigation_cloud)/param/plotter.yaml'/> -->

  <node name='target'
    pkg='circumnavigation_cloud'
    type='target.py' output="screen"/>

  <node name='plotter'
    pkg='circumnavigation_cloud'
    type='plotter.py' output="screen"/>

</launch>
